### ISSUE TITLE
archive: pre github palemoon section

### ISSUE THREAD

**Issue by Roman-Nopantski** [ghacksuserjs/ghacks-user.js#13]
Saved for posterity...

/*** 9996: PALEMOON SPECIFIC ( https://www.palemoon.org/ )
     Full list maintained by Moonchild: https://forum.palemoon.org/viewtopic.php?f=24&t=3357
     If you have issues or questions about any of these, please use the palemoon forums
     NOTE: This section is no longer maintained [after version 10] ***/
// 9996-1: (v25.6+) disable canvas fingerprinting
   // user_pref("canvas.poisondata", true);
// 9996-2: (v25.2+) control HSTS
   // If editing this in about:config PM needs to be fully closed and then restarted
   // NOTE: This is a trade-off between privacy vs security. HSTS was designed to increase
   // security to stop MiTM attacks but can also be misused as a fingerprinting vector, by
   // scrapping previously visited sites. Recommended: security over privacy. Your choice.
   // user_pref("network.stricttransportsecurity.enabled", true);
// 9996-3: (v25.0+) controls whether to ignore an expired state of stapled OCSP responses
   // If set to true, breaks with RFC6066 (like Firefox) and ignores the fact that stapled
   // OCSP responses may be expired. If false (the default) aborts the connection.
   // user_pref("security.ssl.allow_unsafe_ocsp_response", false);
// 9996-4: (v25.6+) Controls whether to completely ignore "autocomplete=off" on login fields
   // user_pref("signon.ignoreAutocomplete", false);
// 9996-5: (v26.0+) read Moonchild's description on the palemoon forum thread linked above
   // user_pref("dom.disable_beforeunload", true);

**Comment by earthlng** [comment]
We could move this into a `palemoon.js` file in the `preferences` folder. A preferences folder would be neat IMO, we could also create a 'deprecated.js'. Or a 'paranoid.js' (or hardened.js) in the future, and things like that. There are 2 approaches for a hardened.js, either a complete copy of the user.js with hardened values and some items commented out, or a separate js file in `preferences` with only the changed items for hardening. In the latter case we would have to move the user.js into the preferences folder and name all files accordingly or the values in user.js would most likely always overwrite the hardened.js values.
fe. 001-palemoon.js, 010-deprecated.js, 090-user.js, 099-hardened.js.
We could also move parts of the user.js into separate files, so end-users could just pick the ones they want. fe. safebrowsing-trackingprotection.js

**Comment by Roman-Nopantski** [comment]
I wouid be ok with a palemoon-user.js just residing in our root repo (maybe move to a sub dir later). But a js of just 5 prefs - what a waste of time. It should be the real deal or nothing, IMO. Someone would have to get a current PM and look up all prefs to see if they exist. I wonder if Moonchild can provide us with a list of all prefs, including hidden ones. He's very approachable, I emailed him way back when this started, and he replied within minutes. And it would be provided as is, not maintained, and anyone can take it to start a PM repo. I have no issues with it being here, but I'll be damned if I'm doing the work for it. Maybe someone from ghacks who is a PM user might have something already - doesn't Tom H run a PM as well - we should as him?

--OT--
Not convinced end users should deal with multiple files and a cfg.(Yes, I want to do a wiki on lock prefs etc). I know you said something about we're not dealing with idiots here - but we should keep things down to a lowest common denominator - KISS principle. We got a lot of exposure from ghacks (almost 1600 v11 zip downloads in 5 days, btw), and some readers have just run with it (despite warnings). Have you seen the questions? "Where does this go?", "How do I undo all this?" etc. And this is now also being used by 12bytes (he forked it) for his article. We have more exposure coming up - Martin is doing another article about the github move. And I can also see him doing another one when the illustrated wiki is all finished. These wikis, I believe, will make it so damn easy, and will encourage more people to try. /end rambling

My thoughts on this, based on my outline readme (privacy vs security on one side balanced with convenience and browser+site functionally on the other), is that we could have three branches.

- The first would be `baby bear` - relies on no addons (so no turning off safebrowsing etc), doesn't mess with any auto-updates, has no site breakage (so no video restrictions), and even leaves things like urlbar suggestions alone (they can do that from options). Basically no loss of browser functionality, or site breakage - that is still a lot of prefs that can be applied.

- Then comes `mama bear` - which is the current one: a balance between sec/privacy vs convenience, loss of some features (GMPs webgl etc) and very little site breakage. The things to troubleshoot are minimal (I think I will reduce that warning list a bit)

- And lastly we have `papa bear` - an all-in balls-to-the-wall super-ultra-hardened setup. Basically turn everything on and test it out. And flip anything where privacy (and "quiet") was favored over security to the other way round (eg enable safebrowsing, TP, and updates for everything) No retention of any data on close, no remembering history during sessions, start in PB mode, block mixed passive .. everything that shoudl be turned on, lets turn it on and test it out. And state it must run with NS in default deny all, and uBlock Origin as bare minimum. Kind of like a default TBB. 

I envisage doing them as two extra branches. I want to get the wiki finished before then, with how to's - such as running concurrent profiles (-no remote switch, or I think their may be an addon). Additionally, I would want them to use different themes per profile for visual clues. (or edit the css or use a stylish script eg I use `.tabbrowser-tab[selected] .tab-label{ color: #96EB41; !important; }` on FT DeepDark theme to change make my active tab's text light pastel green, the others are white.

**That said, your idea holds merit, don't get me wrong.** We also have containers coming up, which may provide some of this functionality.

**Comment by earthlng** [comment]
I don't have much time to properly respond now, but just a quick FYI, the preferences folder doesn't require a cfg file. Create a preferences folder in your profile and put user.js files in there, that's all.
If it's possible we wouldn't have to include that folder in the releases, or only those files which we moved out of the main user.js fe. deprecated, safebrowsing or whatnot.

**Comment by earthlng** [comment]
_"But a js of just 5 prefs - what a waste of time."_ - True, but who is gonna see it here, hidden in a closed issue? If we want more people to contribute to palemoon.js we need to make it more visible.

_almost 1600 v11 zip downloads in 5 days, btw_ - oh wow, I had no idea, judging from the amount of comments. where are all those guys? not even a thank you? damn! xD
I assumed it's like 10-20 people or so using this, lol.

_"Where does this go?", "How do I undo all this?"_ - I'm sorry but there are massive warnings and recommendations (backup, etc) in multiple places in the file and the article itself, so if someone shoots them-self in the foot despite that, I'm not willing to help them. They need to learn it the hard way! RTFM! 

re: papa bear - wait what? we have very different ideas about what a hardened version would look like. enable SB + TP and updates and shit? IMO a hardened version would disable everything.
And maybe let's not use 'bear' in the name, you know... that's kinda tainted since the election ;)

_Additionally, I would want them to use different themes per profile for visual clues._ - who is them? I lost you there.

ps. I'm working on your assigned task for punycode and formfill.expire_days, they should be ready soon-ish

**Comment by Roman-Nopantski** [comment]
- "And maybe let's not use 'bear' in the name, you know... that's kinda tainted since the election" - ROFL - I know huh, I've been pulling out all the election gags for a while now. Did you see the few upset trolls come at me in a recent ghacks article when I called some troll out - I said he was acting like trump, lulz?
- "True, but who is gonna see it here, hidden in a closed issue" - not really high on my list and I dumped PM about a year ago. Too many extensions broken, too divergent, too slow to develop. Good luck to them, but I have better things to do.
- "They need to learn it the hard way" - that may be true, but we still end up wearing the endless issues caused by assuming all end users are smart. I'm being proactive in preventing problems for users, and work for us, and I'm also trying to make this whole thing more accessible for the average person - see readme draft.
- This closed ticket is also locked. Only collaborators and members can write here (I'm playing with a few things to see what the consequences are - am too lazy to go read a FAQ).
- "who is them? I lost you there" - end users, RTFReadme and goals, purpose
- We're getting offtopic - maybe we should start a new discussion topic and dump some ideas in it for branches with a sliding scale. On the topic of enabling SB and TP in the papa bear (pyllyukko's has it left on and they're not silly), I was thinking along the lines of security trumps privacy - not that I think any privacy is actually lost with these except the google real time check for binaries which I would leave disabled. Even though uBlock Origin is IMO superior for this, it cannot hurt to have the extra layer. That was all. All that can be worked out later
- You and I should have some OT discussions elsewhere. Are you happy with something like Pidgin - use a jaber account or whatever its called. Or Signal (then I'll know ya phone number, woo hoo, can sign yu up to the republican party!!). Or email, we can hook up via Martin. Or not, your call.
